src/registrar/forms/domain.py

```diff
@@ -542,6 +541,21 @@ def _field_unchanged(self, field_name) -> bool:
         """
         old_value = self.initial.get(field_name, None)
         new_value = self.cleaned_data.get(field_name, None)
+
```

Comment:
This code was added so the form would submit.  _field_unchanged was comparing self.initial to self.cleaned_data for a select dropdown.  It was comparing the id for self.initial to the object itself in self.cleaned_data, which was comparing an id to a string.  Added some code to handle different use cases here, and resolve the second bug.